### PR TITLE
Add trust self signed certificate

### PR DESCRIPTION
**Pull request by lukasz20p**
Handling self signed certificate.

### CODE REVIEW

File: ribbon-core/src/main/java/com/netflix/client/config/CommonClientConfigKey.java [hunk 1]
```diff
@@ -134,8 +134,15 @@
     
     // if this is a secure rest client, must we use client auth too?    
     public static final IClientConfigKey<Boolean> IsClientAuthRequired = new CommonClientConfigKey<Boolean>("IsClientAuthRequired"){};
-    
+
+    /**
+     * Use {@link #TrustSelfSignedCertificate}
+     */
+    @Deprecated
```
Comment (lukasz20p) [inline ribbon-core/src/main/java/com/netflix/client/config/CommonClientConfigKey.java]: This field is no more used. Only deprecated RestClient used it.

File: ribbon-core/src/main/java/com/netflix/client/ssl/SslContext.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.netflix.client.ssl;
+
+import javax.net.ssl.SSLContext;
+
+public interface SslContext {
```
Comment (lukasz20p) [inline ribbon-core/src/main/java/com/netflix/client/ssl/SslContext.java]: Interface used for getContext

File: ribbon-core/src/main/java/com/netflix/client/ssl/TrustSelfSignedCertificateFactory.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+            return sslcontext;
+        } catch (NoSuchAlgorithmException e) {
+            throw new ClientSslContextFactoryException(String.format("Failed to create an SSL context that supports algorithm %s: %s", SOCKET_ALGORITHM, e.getMessage()), e);
+        } catch (KeyManagementException e) {
+            throw new ClientSslContextFactoryException(String.format("Failed to initialize an SSL context: %s", e.getMessage()), e);
+        }
+    }
+
+    private class TrustSelfSignedStrategy implements X509TrustManager {
```
Comment (lukasz20p) [inline ribbon-core/src/main/java/com/netflix/client/ssl/TrustSelfSignedCertificateFactory.java]: We trust self signed certificate.

**Comment by Zenedith** [comment]
:+1:

**Comment by przemeq** [comment]
+1

**Comment by mhawrylczak** [comment]
:+1:

**Pull request closed without merging**